pkg/kubectl/cmd/top.go

```diff
@@ -0,0 +1,166 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"strings"
+
+	"github.com/renstrom/dedent"
+	"github.com/spf13/cobra"
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/unversioned"
+	client "k8s.io/kubernetes/pkg/client/unversioned"
+	"k8s.io/kubernetes/pkg/kubectl"
+	cmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+	"k8s.io/kubernetes/pkg/kubectl/metricsutil"
+)
+
+// TopOptions is the start of the data required to perform the operation. As new fields are added, add them here instead of
+// referencing the cmd.Flags()
+type TopOptions struct {
+}
+
+var (
+	topLong = dedent.Dedent(`
+		Display Resource (CPU/Memory/Storage) usage of nodes or pods.
+
+		The top command allows you to see the resource consumption of the nodes or pods.
+		It downloads the usage metrics of a given resource (node/pod) via the Resource Metrics API.`)
+
+	topExample = dedent.Dedent(`
+		  # Show metrics for all nodes
+		  kubectl top node
```

Comment:
Both singular and plural forms work - that's what I've seen in other commands (done by `kubectl.ResourceAliases(validArgs)` below)
But you're right, it would look better here in plural and would be consistent with other commands' examples.